jetty-websocket/websocket-servlet/src/main/java/org/eclipse/jetty/websocket/servlet/WebSocketServlet.java

```diff
@@ -53,167 +51,242 @@
             if (max != null)
-            {
-                factory.setDefaultMaxBinaryMessageSize(Long.parseLong(max));
-            }
+                customizer.setMaxBinaryMessageSize(Long.parseLong(max));
 
             max = getInitParameter("inputBufferSize");
             if (max != null)
-            {
-                factory.setDefaultInputBufferSize(Integer.parseInt(max));
-            }
+                customizer.setInputBufferSize(Integer.parseInt(max));
 
             max = getInitParameter("outputBufferSize");
             if (max != null)
-            {
-                factory.setDefaultOutputBufferSize(Integer.parseInt(max));
-            }
+                customizer.setOutputBufferSize(Integer.parseInt(max));
 
-            max = getInitParameter("maxAllowedFrameSize");
+            max = getInitParameter("maxFrameSize");
+            if (max==null)
+                max = getInitParameter("maxAllowedFrameSize");
             if (max != null)
-            {
-                factory.setDefaultMaxAllowedFrameSize(Long.parseLong(max));
-            }
+                customizer.setMaxFrameSize(Long.parseLong(max));
 
             String autoFragment = getInitParameter("autoFragment");
             if (autoFragment != null)
-            {
-                factory.setAutoFragment(Boolean.parseBoolean(autoFragment));
-            }
-
-            List<WebSocketServletFrameHandlerFactory> factories = (List<WebSocketServletFrameHandlerFactory>)ctx.getAttribute(
-                WebSocketServletFrameHandlerFactory.ATTR_HANDLERS);
-            if (factories != null)
-                factories.forEach(factory::addFrameHandlerFactory);
+                customizer.setAutoFragment(Boolean.parseBoolean(autoFragment));
 
-            configure(factory); // Let user modify factory
-
-            ctx.setAttribute(WebSocketCreatorMapping.class.getName(), factory);
+            configure(customizer); // Let user modify customizer prior after init params
         }
         catch (Throwable x)
         {
             throw new ServletException(x);
         }
     }
 
-    /**
-     * @see javax.servlet.http.HttpServlet#service(javax.servlet.http.HttpServletRequest, javax.servlet.http.HttpServletResponse)
-     */
     @Override
-    protected void service(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException
+    protected void service(HttpServletRequest req, HttpServletResponse resp)
+        throws ServletException, IOException
     {
-        // Since this is a filter, we need to be smart about determining the target path.
-        // We should rely on the Container for stripping path parameters and its ilk before
-        // attempting to match a specific mapped websocket creator.
-        String target = request.getServletPath();
-        if (request.getPathInfo() != null)
+        // Typically this servlet is used together with the WebSocketUpgradeFilter,
```

Comment:
Environments with no WSUF and only a WebSocketServlet is very common.

Reply:
However, now both SCIs for Jetty and Javax will install a filter, so it should become very uncommon.  However, the use-case is still supported and I'll destress the `Typically`